### ISSUE TITLE
Investigate recursion loop in Debate.neg_team

### ISSUE THREAD

**Issue by czlee** [czlee/tabbycat#335]
There's some sort of recursion loop, clearly not infinite because the page does eventually load, but this seems concerning: 

![image](https://cloud.githubusercontent.com/assets/1725499/16679142/2103b17c-449a-11e6-9e54-a8c687520d86.png)

I think __get__ messes up the profiler, so this might be a mistake in profiling, but it seems to me that `Debate.neg_team` shouldn't be calling itself over and over.

**Comment by czlee** [comment]
Yeah, I think it just doesn't know how to follow `__get__`. This is more enlightening:

![image](https://cloud.githubusercontent.com/assets/1725499/16679512/fd0d2034-449c-11e6-9e1c-6ce176c2f26a.png)

Worth noting: `aff_team` and `neg_team` both use `get()`, as do `aff_dt` and `neg_dt`, so I think they incur database hits each time, _i.e._ they don't return QuerySet. However, they also use select_related:

``` python
    @cached_property
    def aff_dt(self):
        return DebateTeam.objects.select_related('team', 'team__institution').get(
                debate=self, position=DebateTeam.POSITION_AFFIRMATIVE)

    @cached_property
    def neg_dt(self):
        return DebateTeam.objects.select_related('team', 'team__institution').get(
                debate=self, position=DebateTeam.POSITION_NEGATIVE)
```

I'm not sure to what extent this helps or harms performance.

**Comment by czlee** [comment]
Actually, it should probably be fairly obvious that this is harmful—every time it looks up an `aff_dt` that hasn't already been looked up, it has to run that query that selects all items in `DebateTeam` and the corresponding `Team` and `Institution`.

**Comment by czlee** [comment]
All a red herring, the culprit here was an inefficient `Team.seen()`, which was fixed in b7dc324.
